prow/pjutil/pjutil.go

```diff
@@ -185,3 +185,32 @@ func EnvForSpec(spec kube.ProwJobSpec) map[string]string {
 	env["PULL_PULL_SHA"] = spec.Refs.Pulls[0].SHA
 	return env
 }
+
+// PartitionProwJobs separates the provided prowjobs into pending and non-pending
+// and returns them inside channels so that they can be consumed in parallel
+// by different goroutines. Controller loops need to handle pending jobs first
+// so they can conform to maximum concurrency requirements that different jobs
+// may have.
+func PartitionProwJobs(pjs []kube.ProwJob) (pending, nonPending chan kube.ProwJob) {
```

Comment:
Name this something more useful such as `PartitionPending`. Can you put `(pending, nonPending <-chan kube.ProwJob)`?